src/modem/mod.rs

```diff
@@ -186,10 +192,16 @@ impl<'c, P: ModemPower> Modem<'c, P> {
         commands
             .run(cmee::ConfigureCMEErrors(CMEErrorMode::Numeric))
             .await?;
-        commands
-            .run(cnmp::SetNetworkMode(config.network_mode))
-            .await?;
-        commands.run(cmnb::SetNbMode(config.nb_mode)).await?;
+
+        if config.network_mode == NetworkMode::Automatic {
+            self.automatic_registration = true;
```

Comment:
This feels off to me. The NetworkMode enum maps 1-1 with the modem AT command API, but here we are hijacking the variant `Automatic` to mean something else, i.e. automatic mode selection by the driver, instead of automatic mode selection by the modem).